src/components/views/messages/RoomAvatarEvent.js

```diff
@@ -75,8 +75,8 @@ module.exports = React.createClass({
                     {
                         'img': () =>
                             <AccessibleButton key="avatar" className="mx_RoomAvatarEvent_avatar"
-                                onClick={this.onAvatarClick.bind(this, name)}>
-                                <BaseAvatar width={14} height={14} url={url} name={name} />
+                                onClick={this.onAvatarClick}>
+                                <RoomAvatar width={14} height={14} room={room} oobData={oobData} />
```

Comment:
I think I've figured out what's going on here: we're using RoomAvatar so it gets the fallback avatar right, but (cheekily) using oobData to override the URL so it's the avatar from this event not whatever the current one is? A comment to explain this might be good as this took me a good few minutes to think through.

Reply:
I guess that now you say it, it can have undesired behaviour so I will back it out.
If an invalid avatar was set, then a valid one, instead of falling back to the default-letter-avatar it'd fallback to the current valid one.

Reply:
Actually, RoomAvatar says it can be used without a room, with just an oobData.avatarUrl so that'd be the cleanest IMO